lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1768,9 +1768,9 @@ def _shade_colors(self, color, normals, lightsource=None):
             # chosen for backwards-compatibility
             lightsource = LightSource(azdeg=225, altdeg=19.4712)
 
-        shade = np.array([np.dot(n / proj3d.mod(n), lightsource.direction)
-                          if proj3d.mod(n) else np.nan
-                          for n in normals])
+        with np.errstate(invalid="ignore"):
+            shade = ((normals / np.linalg.norm(normals, axis=1, keepdims=True))
+                     @ lightsource.direction)
```

Comment:
OK, I couldn't even google this one.  What does the `@` operator do?

Reply:
That‘s a new operator introduced in python 3. Numpy uses it for matrix multiplication. `a @ b` is the same as `np.dot(a, b)`, just more readable.